beacon_node/beacon_chain/src/fetch_blobs.rs

```diff
@@ -0,0 +1,282 @@
+                            );
+                            return;
+                        }
+                    };
+
+                    // Check indices from cache before sending the columns, to make sure we don't
+                    // publish components already seen on gossip.
+                    let all_data_columns_iter = all_data_columns.clone().into_iter();
+                    let data_columns_to_publish = match chain_cloned
+                        .data_availability_checker
+                        .imported_custody_column_indexes(&block_root)
+                    {
+                        None => Either::Left(all_data_columns_iter),
+                        Some(imported_columns_indices) => Either::Right(
+                            all_data_columns_iter
+                                .filter(move |d| !imported_columns_indices.contains(&d.index())),
+                        ),
+                    }
+                    .collect::<Vec<_>>();
+
+                    if let Err(e) = data_columns_sender.try_send(all_data_columns) {
+                        error!(logger, "Failed to send computed data columns"; "error" => ?e);
+                    };
+
+                    let is_supernode = chain_cloned
+                        .data_availability_checker
+                        .get_custody_columns_count()
+                        == spec.number_of_columns;
+                    if is_supernode && !data_columns_to_publish.is_empty() {
+                        publish_fn(BlobsOrDataColumns::DataColumns(data_columns_to_publish));
+                    }
+                },
+                "compute_data_columns",
+            )
+            .ok_or(FetchEngineBlobError::RuntimeShutdown)?;
+
+        Some(data_columns_receiver)
+    } else {
+        if num_fetched_blobs == 0 {
+            debug!(
+                chain.log,
+                "No blobs fetched from the EL";
+                "num_expected_blobs" => num_expected_blobs,
+            );
+            inc_counter(&metrics::BLOBS_FROM_EL_MISS_TOTAL);
+            return Ok(());
+        }
+
+        inc_counter(&metrics::BLOBS_FROM_EL_HIT_TOTAL);
+
+        let all_blobs = fixed_blob_sidecar_list.clone();
+        let all_blobs_iter = all_blobs.into_iter().flat_map(|b| b.clone());
+
+        let blobs_to_publish = match chain
+            .data_availability_checker
+            .imported_blob_indexes(&block_root)
+        {
+            None => Either::Left(all_blobs_iter),
+            Some(imported_blob_indices) => Either::Right(
+                all_blobs_iter.filter(move |b| !imported_blob_indices.contains(&b.index())),
+            ),
+        };
+
+        publish_fn(BlobsOrDataColumns::Blobs(
+            blobs_to_publish.collect::<Vec<_>>(),
+        ));
+
+        None
+    };
+
+    debug!(
+        chain.log,
+        "Blobs from EL - start processing";
+        "num_fetched_blobs" => num_fetched_blobs,
+    );
+
+    chain
+        .process_engine_blobs(
```

Comment:
just to make sure we don't forget - we should process the returned `AvailabilityProcessingStatus` and recompute head

Reply:
Fixed in https://github.com/sigp/lighthouse/pull/6268/commits/7977999b4a74db5fd9942d0463bcc642550feed2